src/addrdb.cpp

```diff
@@ -36,7 +37,7 @@ template <typename Data>
 bool SerializeFileDB(const std::string& prefix, const fs::path& path, const Data& data)
 {
     // Generate random temporary filename
-    unsigned short randv = 0;
+    uint16_t randv = 0;
     GetRandBytes((unsigned char*)&randv, sizeof(randv));
     std::string tmpfn = strprintf("%s.%04x", prefix, randv);
```

Comment:
This is okay only because `strprintf` has some type magic.

Reply:
Yes, this isn't a problem, `strprintf` is type safe. There is never a need for size specifiers.